api/v1alpha1/securitygroup_types.go

```diff
@@ -26,40 +26,88 @@ type RuleDirection string
 // +kubebuilder:validation:MaxLength:=16
 type Protocol string
 
+const (
+	ProtocolANY       Protocol = "any"
+	ProtocolAH        Protocol = "ah"
+	ProtocolDCCP      Protocol = "dccp"
+	ProtocolEGP       Protocol = "egp"
+	ProtocolESP       Protocol = "esp"
+	ProtocolGRE       Protocol = "gre"
+	ProtocolICMP      Protocol = "icmp"
+	ProtocolICMPV6    Protocol = "icmpv6"
+	ProtocolIGMP      Protocol = "igmp"
+	ProtocolIPIP      Protocol = "ipip"
+	ProtocolIPV6ENCAP Protocol = "ipv6-encap"
+	ProtocolIPV6FRAG  Protocol = "ipv6-frag"
+	ProtocolIPV6ICMP  Protocol = "ipv6-icmp"
+	ProtocolIPV6NONXT Protocol = "ipv6-nonxt"
+	ProtocolIPV6OPTS  Protocol = "ipv6-opts"
+	ProtocolIPV6ROUTE Protocol = "ipv6-route"
+	ProtocolOSPF      Protocol = "ospf"
+	ProtocolPGM       Protocol = "pgm"
+	ProtocolRSVP      Protocol = "rsvp"
+	ProtocolSCTP      Protocol = "sctp"
+	ProtocolTCP       Protocol = "tcp"
+	ProtocolUDP       Protocol = "udp"
+	ProtocolUDPLITE   Protocol = "udplite"
+	ProtocolVRRP      Protocol = "vrrp"
+)
+
 // +kubebuilder:validation:Enum:=IPv4;IPv6
 // +kubebuilder:validation:MinLength:=1
-// +kubebuilder:validation:MaxLength:=16
+// +kubebuilder:validation:MaxLength:=4
```

Comment:
I'm curious: what does MaxLength do when the field is an enum?

Reply:
I also wondered this. It's possible that even though it's technically redundant for validation purposes, it may still be used in the runtime complexity calculations? I'll bet @joelspeed would know.

Reply:
For runtime complexity, the openapi schema shows that the field is an enum, so the estimated worst case is based on the maximum length of the enum values. You don't actually need a maxlength on an enum WRT CEL, but it also cannot hurt to include it

Reply:
I'll remove both length rules